plugin/trino-hive/src/main/java/io/trino/plugin/hive/metastore/glue/GlueMetastoreModule.java

```diff
@@ -128,8 +130,9 @@ public static GlueClient createGlueClient(GlueHiveMetastoreConfig config, OpenTe
                         .setCaptureExperimentalSpanAttributes(true)
                         .setRecordIndividualHttpError(true)
                         .build().newExecutionInterceptor())
-                .retryPolicy(retry -> retry
-                        .numRetries(config.getMaxGlueErrorRetries())));
+                .retryStrategy(retryBuilder -> retryBuilder
+                        .retryOnException(throwable -> Throwables.getRootCause(throwable) instanceof ConcurrentModificationException)
```

Comment:
I don't think this is correct. This retry is in the Glue client and `software.amazon.awssdk.services.glue.model.ConcurrentModificationException` is a `GlueException`, so it should be thrown directly from the client.

The original retry code for Glue v1 `dropTable` and associated test seems to be based on a stack trace after the failure was wrapped in `TrinoException`, which wouldn't be the case. Although it would be correct for the later added `updateTableStatistics` as that retry happens outside of the wrapping. The test `TestHiveConcurrentModificationGlueMetastore` should be updated.

Ideally, we can find an actual exception from the Glue service to validate this.

Reply:
I think it is fine, looking at v2 client retries, it uses `AwsRetryStrategy#configureStrategy` which uses same mechanism, adding some default handling of `AwsServiceException` classes, so adding handling of another children of this class (`ConcurrentModificationException`) would just extend this behavior

But `Throwables.getRootCause` is not needed here, as it can be checked with `instanceof` directly, as we are not wrapping it with trino exception at this point

Reply:
When it comes to v1, it is analogical, `com.amazonaws.retry.PredefinedRetryPolicies` also check if exception is `AmazonServiceException` when performing retries, so its not only using it externally to the client, but also internally, so its perfectly fine to check it in lambda, but again, we don't need to check root cause but just plain instanceof should do it

Reply:
there is article on that here, that external retries shouldn't be done, but retry customizers should be used instead:
https://docs.aws.amazon.com/codeguru/detector-library/java/aws-custom-retries/

Reply:
I changed second commit `Do not rely on Failsafe in glue metastore` to test if retry policy is set up correctly on glue client, instead of proxying method with throwing exception, as it doesn't work anymore due to exception being handled deeper

Reply:
@electrum can I have re-review after latest changes? I'd like to proceed with this PR